### PR TITLE
Game manager

### PR DESCRIPTION
**Pull request by milosmilosevic13**
Added game manager that takes care of spawning enemies and monitors game state, ie. if the game should end and with what ending.

### CODE REVIEW

File: source/dagger/gameplay/team_game/game_manager.cpp [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+void ancient_defenders::GameManagerSystem::WindDown()
+{
+    Engine::Dispatcher().sink<NextFrame>().disconnect<&GameManagerSystem::OnEndOfFrame>(this);
+}
+
+void ancient_defenders::GameManagerSystem::Run()
+{
+    auto player = Engine::GetDefaultResource<PlayerInfo>();
+    if (player == nullptr) return;
```
Comment (DPrivalov) [inline source/dagger/gameplay/team_game/game_manager.cpp:27]: same as in previous pull request: {}

File: source/dagger/gameplay/team_game/game_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+
+    std::random_device dev;
+    std::mt19937 rng(dev());
+
+    std::uniform_int_distribution<std::mt19937::result_type> roll10(0, 9);
+
+    auto spawnCode = roll10(rng);
+
+    if (spawnCode < 4) { // 40% chance to spawn a little golem
```
Comment (DPrivalov) [inline source/dagger/gameplay/team_game/game_manager.cpp:56]: can be good to put these values into some const sequence (vector)

File: source/dagger/gameplay/team_game/game_manager.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        while (itA != viewA.end()) {
+            AnimatorStop(reg.get<Animator>(*itA));
+            itA++;
+        }
+
+        
+        ancient_defenders::SetupEndScreen(Engine::Instance(), m_victory?true:false);
+
+        m_victory = m_defeat = false;
+        /*
```
Comment (DPrivalov) [inline source/dagger/gameplay/team_game/game_manager.cpp]: delete this comment if you don't need this

**Review by DPrivalov — APPROVED** [review]
When u start merging do this:

merge team's main branch into yours first, push it. 
then merge your changes from github into team's main. (it should not have conflicts then)

File: source/dagger/gameplay/team_game/game_manager.cpp (hunk 3 above)
Comment (milosmilosevic13) [reply source/dagger/gameplay/team_game/game_manager.cpp]: It was deleted locally, I just didn't push the change yet.

**Pull request merged**